src/seedu/addressbook/data/person/Person.java

```diff
@@ -81,7 +82,25 @@ public int hashCode() {
 
     @Override
     public String toString() {
-        return getAsTextShowAll();
+        return getPrintableString(name, phone, email, address);
+        //return getAsTextShowAll();
+    }
+
+    public String getPrintableString(Printable... printables) {
+        final StringBuilder builder = new StringBuilder();
+        final String detailIsPrivate = "(private) ";
+
+        for(Printable p : printables){
+            builder.append(p.getPrintableString());
+            builder.append(", ");
+        }
+
+        builder.append(" Tags: ");
+        for (Tag tag : getTags()) {
+            builder.append(tag);
```

Comment:
can try out doing this using StringJoiner which makes your code "prettier" 
Read about it [here](https://stackoverflow.com/questions/27522563/why-stringjoiner-when-we-already-have-stringbuilder)